lib/rspec/matchers/built_in/match_with_captures.rb

```diff
@@ -0,0 +1,62 @@
+module RSpec
+  module Matchers
+    module BuiltIn
+      # @api private
+      # Extends Match behavior to also match against expected captures
+      module WithCaptures
+        # @api private
+        # @return [String]
+        def description
+          "#{super} with captures #{surface_descriptions_in(@expected_captures).inspect}"
+        end
+
+      private
+
+        def match(expected, actual)
+          (match = super) && values_match?(@expected_captures, actual_captures(match))
+        end
+
+        def actual_captures(match)
+          ReliableMatchData.new(match).captures
+        end
+
+        # @api private
+        # Used to wrap match data and make it reliable for 1.8.7
+        class ReliableMatchData
+          def initialize(match_data)
+            @match_data = match_data || NullMatchData.new
```

Comment:
`if RUBY_VERSION == '1.8.7'` is sufficient, you can't install rspec on anything lower than 1.8.7 anyway